[2025-03-26T13:50:02.477+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-03-26T13:50:02.511+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_pipeline.transformation manual__2025-03-26T13:33:39.761724+00:00 [queued]>
[2025-03-26T13:50:02.521+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_pipeline.transformation manual__2025-03-26T13:33:39.761724+00:00 [queued]>
[2025-03-26T13:50:02.524+0000] {taskinstance.py:2306} INFO - Starting attempt 6 of 7
[2025-03-26T13:50:02.558+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): transformation> on 2025-03-26 13:33:39.761724+00:00
[2025-03-26T13:50:02.565+0000] {standard_task_runner.py:63} INFO - Started process 1120 to run task
[2025-03-26T13:50:02.568+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'etl_pipeline', 'transformation', 'manual__2025-03-26T13:33:39.761724+00:00', '--job-id', '9', '--raw', '--subdir', 'DAGS_FOLDER/etl_pipeline.py', '--cfg-path', '/tmp/tmpg9rgz3m5']
[2025-03-26T13:50:02.570+0000] {standard_task_runner.py:91} INFO - Job 9: Subtask transformation
[2025-03-26T13:50:02.647+0000] {task_command.py:426} INFO - Running <TaskInstance: etl_pipeline.transformation manual__2025-03-26T13:33:39.761724+00:00 [running]> on host 9d8a997eda77
[2025-03-26T13:50:02.890+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='lobobranco' AIRFLOW_CTX_DAG_ID='etl_pipeline' AIRFLOW_CTX_TASK_ID='transformation' AIRFLOW_CTX_EXECUTION_DATE='2025-03-26T13:33:39.761724+00:00' AIRFLOW_CTX_TRY_NUMBER='6' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-26T13:33:39.761724+00:00'
[2025-03-26T13:50:02.892+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-03-26T13:50:02.968+0000] {etl_pipeline.py:76} INFO - Transforming data from file: s3://raw/playlist_20250326_133348.json
[2025-03-26T13:50:02.970+0000] {baseoperator.py:400} WARNING - SparkSubmitOperator.execute cannot be called outside TaskInstance!
[2025-03-26T13:50:02.984+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-03-26T13:50:02.985+0000] {spark_submit.py:328} ERROR - Failed to decode base64 keytab: Incorrect padding
[2025-03-26T13:50:02.986+0000] {spark_submit.py:305} INFO - Could not load connection string spark_default, defaulting to spark://spark-master:7077
[2025-03-26T13:50:02.988+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=2 --conf spark.jars=/opt/spark_job/aws-java-sdk-bundle-1.12.262.jar,/opt/spark_job/jars/hadoop-aws-3.3.4.jar --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=admin --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem --conf spark.hadoop.fs.s3a.path.style.access=true --principal *** --name arrow-spark --verbose --deploy-mode client /spark_job/data_transformation.py s3://raw/playlist_20250326_133348.json
[2025-03-26T13:50:03.106+0000] {spark_submit.py:645} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-03-26T13:50:04.871+0000] {spark_submit.py:645} INFO - Using properties file: null
[2025-03-26T13:50:04.993+0000] {spark_submit.py:645} INFO - Parsed arguments:
[2025-03-26T13:50:04.994+0000] {spark_submit.py:645} INFO - master                  spark://spark-master:7077
[2025-03-26T13:50:04.994+0000] {spark_submit.py:645} INFO - remote                  null
[2025-03-26T13:50:04.995+0000] {spark_submit.py:645} INFO - deployMode              client
[2025-03-26T13:50:04.995+0000] {spark_submit.py:645} INFO - executorMemory          2g
[2025-03-26T13:50:04.995+0000] {spark_submit.py:645} INFO - executorCores           2
[2025-03-26T13:50:04.996+0000] {spark_submit.py:645} INFO - totalExecutorCores      null
[2025-03-26T13:50:04.996+0000] {spark_submit.py:645} INFO - propertiesFile          null
[2025-03-26T13:50:04.997+0000] {spark_submit.py:645} INFO - driverMemory            null
[2025-03-26T13:50:04.998+0000] {spark_submit.py:645} INFO - driverCores             null
[2025-03-26T13:50:04.998+0000] {spark_submit.py:645} INFO - driverExtraClassPath    null
[2025-03-26T13:50:04.999+0000] {spark_submit.py:645} INFO - driverExtraLibraryPath  null
[2025-03-26T13:50:04.999+0000] {spark_submit.py:645} INFO - driverExtraJavaOptions  null
[2025-03-26T13:50:05.000+0000] {spark_submit.py:645} INFO - supervise               false
[2025-03-26T13:50:05.000+0000] {spark_submit.py:645} INFO - queue                   null
[2025-03-26T13:50:05.001+0000] {spark_submit.py:645} INFO - numExecutors            null
[2025-03-26T13:50:05.002+0000] {spark_submit.py:645} INFO - files                   null
[2025-03-26T13:50:05.002+0000] {spark_submit.py:645} INFO - pyFiles                 null
[2025-03-26T13:50:05.003+0000] {spark_submit.py:645} INFO - archives                null
[2025-03-26T13:50:05.003+0000] {spark_submit.py:645} INFO - mainClass               null
[2025-03-26T13:50:05.004+0000] {spark_submit.py:645} INFO - primaryResource         file:/spark_job/data_transformation.py
[2025-03-26T13:50:05.005+0000] {spark_submit.py:645} INFO - name                    arrow-spark
[2025-03-26T13:50:05.006+0000] {spark_submit.py:645} INFO - childArgs               [s3://raw/playlist_20250326_133348.json]
[2025-03-26T13:50:05.006+0000] {spark_submit.py:645} INFO - jars                    /opt/spark_job/aws-java-sdk-bundle-1.12.262.jar,/opt/spark_job/jars/hadoop-aws-3.3.4.jar
[2025-03-26T13:50:05.011+0000] {spark_submit.py:645} INFO - packages                null
[2025-03-26T13:50:05.012+0000] {spark_submit.py:645} INFO - packagesExclusions      null
[2025-03-26T13:50:05.014+0000] {spark_submit.py:645} INFO - repositories            null
[2025-03-26T13:50:05.015+0000] {spark_submit.py:645} INFO - verbose                 true
[2025-03-26T13:50:05.015+0000] {spark_submit.py:645} INFO - 
[2025-03-26T13:50:05.016+0000] {spark_submit.py:645} INFO - Spark properties used, including those specified through
[2025-03-26T13:50:05.023+0000] {spark_submit.py:645} INFO - --conf and those from the properties file null:
[2025-03-26T13:50:05.024+0000] {spark_submit.py:645} INFO - (spark.executor.cores,2)
[2025-03-26T13:50:05.025+0000] {spark_submit.py:645} INFO - (spark.executor.memory,2g)
[2025-03-26T13:50:05.026+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.access.key,*********(redacted))
[2025-03-26T13:50:05.026+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.endpoint,http://minio:9000)
[2025-03-26T13:50:05.027+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.impl,org.apache.hadoop.fs.s3a.S3AFileSystem)
[2025-03-26T13:50:05.027+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.path.style.access,true)
[2025-03-26T13:50:05.028+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.secret.key,*********(redacted))
[2025-03-26T13:50:05.028+0000] {spark_submit.py:645} INFO - (spark.jars,/opt/spark_job/aws-java-sdk-bundle-1.12.262.jar,/opt/spark_job/jars/hadoop-aws-3.3.4.jar)
[2025-03-26T13:50:05.028+0000] {spark_submit.py:645} INFO - (spark.master,spark://spark-master:7077)
[2025-03-26T13:50:05.029+0000] {spark_submit.py:645} INFO - 
[2025-03-26T13:50:05.029+0000] {spark_submit.py:645} INFO - 
[2025-03-26T13:50:05.454+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:05 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-03-26T13:50:05.632+0000] {spark_submit.py:645} INFO - Main class:
[2025-03-26T13:50:05.632+0000] {spark_submit.py:645} INFO - org.apache.spark.deploy.PythonRunner
[2025-03-26T13:50:05.633+0000] {spark_submit.py:645} INFO - Arguments:
[2025-03-26T13:50:05.633+0000] {spark_submit.py:645} INFO - file:/spark_job/data_transformation.py
[2025-03-26T13:50:05.633+0000] {spark_submit.py:645} INFO - null
[2025-03-26T13:50:05.634+0000] {spark_submit.py:645} INFO - s3://raw/playlist_20250326_133348.json
[2025-03-26T13:50:05.635+0000] {spark_submit.py:645} INFO - Spark config:
[2025-03-26T13:50:05.635+0000] {spark_submit.py:645} INFO - (spark.app.name,arrow-spark)
[2025-03-26T13:50:05.635+0000] {spark_submit.py:645} INFO - (spark.app.submitTime,1742997005615)
[2025-03-26T13:50:05.636+0000] {spark_submit.py:645} INFO - (spark.executor.cores,2)
[2025-03-26T13:50:05.636+0000] {spark_submit.py:645} INFO - (spark.executor.memory,2g)
[2025-03-26T13:50:05.636+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.access.key,*********(redacted))
[2025-03-26T13:50:05.637+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.endpoint,http://minio:9000)
[2025-03-26T13:50:05.637+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.impl,org.apache.hadoop.fs.s3a.S3AFileSystem)
[2025-03-26T13:50:05.637+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.path.style.access,true)
[2025-03-26T13:50:05.638+0000] {spark_submit.py:645} INFO - (spark.hadoop.fs.s3a.secret.key,*********(redacted))
[2025-03-26T13:50:05.638+0000] {spark_submit.py:645} INFO - (spark.jars,file:/opt/spark_job/aws-java-sdk-bundle-1.12.262.jar,file:/opt/spark_job/jars/hadoop-aws-3.3.4.jar)
[2025-03-26T13:50:05.638+0000] {spark_submit.py:645} INFO - (spark.kerberos.principal,***)
[2025-03-26T13:50:05.638+0000] {spark_submit.py:645} INFO - (spark.master,spark://spark-master:7077)
[2025-03-26T13:50:05.639+0000] {spark_submit.py:645} INFO - (spark.repl.local.jars,file:/opt/spark_job/aws-java-sdk-bundle-1.12.262.jar,file:/opt/spark_job/jars/hadoop-aws-3.3.4.jar)
[2025-03-26T13:50:05.639+0000] {spark_submit.py:645} INFO - (spark.submit.deployMode,client)
[2025-03-26T13:50:05.639+0000] {spark_submit.py:645} INFO - (spark.submit.pyFiles,)
[2025-03-26T13:50:05.640+0000] {spark_submit.py:645} INFO - Classpath elements:
[2025-03-26T13:50:05.640+0000] {spark_submit.py:645} INFO - file:/opt/spark_job/aws-java-sdk-bundle-1.12.262.jar
[2025-03-26T13:50:05.641+0000] {spark_submit.py:645} INFO - file:/opt/spark_job/jars/hadoop-aws-3.3.4.jar
[2025-03-26T13:50:05.641+0000] {spark_submit.py:645} INFO - 
[2025-03-26T13:50:05.641+0000] {spark_submit.py:645} INFO - 
[2025-03-26T13:50:05.642+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:05 WARN DependencyUtils: Local jar /opt/spark_job/aws-java-sdk-bundle-1.12.262.jar does not exist, skipping.
[2025-03-26T13:50:05.642+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:05 WARN DependencyUtils: Local jar /opt/spark_job/jars/hadoop-aws-3.3.4.jar does not exist, skipping.
[2025-03-26T13:50:06.957+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:06 INFO SparkContext: Running Spark version 3.5.5
[2025-03-26T13:50:06.958+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:06 INFO SparkContext: OS info Linux, 5.15.153.1-microsoft-standard-WSL2, amd64
[2025-03-26T13:50:06.959+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:06 INFO SparkContext: Java version 17.0.14
[2025-03-26T13:50:06.990+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:06 INFO ResourceUtils: ==============================================================
[2025-03-26T13:50:06.991+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:06 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-03-26T13:50:06.992+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:06 INFO ResourceUtils: ==============================================================
[2025-03-26T13:50:06.993+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:06 INFO SparkContext: Submitted application: Minio Integration with PySpark
[2025-03-26T13:50:07.015+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 2, script: , vendor: , memory -> name: memory, amount: 2048, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-03-26T13:50:07.028+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO ResourceProfile: Limiting resource is cpus at 2 tasks per executor
[2025-03-26T13:50:07.030+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-03-26T13:50:07.087+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SecurityManager: Changing view acls to: ***
[2025-03-26T13:50:07.088+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SecurityManager: Changing modify acls to: ***
[2025-03-26T13:50:07.088+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SecurityManager: Changing view acls groups to:
[2025-03-26T13:50:07.089+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SecurityManager: Changing modify acls groups to:
[2025-03-26T13:50:07.090+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-03-26T13:50:07.357+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO Utils: Successfully started service 'sparkDriver' on port 40787.
[2025-03-26T13:50:07.399+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SparkEnv: Registering MapOutputTracker
[2025-03-26T13:50:07.440+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SparkEnv: Registering BlockManagerMaster
[2025-03-26T13:50:07.465+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-03-26T13:50:07.466+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-03-26T13:50:07.470+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-03-26T13:50:07.494+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-6c1f1c0a-2c30-429d-b4ac-1290f3cdbc91
[2025-03-26T13:50:07.510+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-03-26T13:50:07.529+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-03-26T13:50:07.743+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-03-26T13:50:07.896+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-03-26T13:50:07.967+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 ERROR SparkContext: Failed to add file:/opt/spark_job/aws-java-sdk-bundle-1.12.262.jar to Spark environment
[2025-03-26T13:50:07.968+0000] {spark_submit.py:645} INFO - java.io.FileNotFoundException: Jar /opt/spark_job/aws-java-sdk-bundle-1.12.262.jar not found
[2025-03-26T13:50:07.968+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2025-03-26T13:50:07.969+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2025-03-26T13:50:07.969+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2025-03-26T13:50:07.969+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2025-03-26T13:50:07.970+0000] {spark_submit.py:645} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-26T13:50:07.970+0000] {spark_submit.py:645} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-26T13:50:07.970+0000] {spark_submit.py:645} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-26T13:50:07.971+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2025-03-26T13:50:07.971+0000] {spark_submit.py:645} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-03-26T13:50:07.972+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-03-26T13:50:07.972+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-03-26T13:50:07.973+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-03-26T13:50:07.973+0000] {spark_submit.py:645} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-03-26T13:50:07.973+0000] {spark_submit.py:645} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-03-26T13:50:07.974+0000] {spark_submit.py:645} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-03-26T13:50:07.974+0000] {spark_submit.py:645} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-26T13:50:07.975+0000] {spark_submit.py:645} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-03-26T13:50:07.976+0000] {spark_submit.py:645} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-03-26T13:50:07.976+0000] {spark_submit.py:645} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-03-26T13:50:07.977+0000] {spark_submit.py:645} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-26T13:50:07.977+0000] {spark_submit.py:645} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-26T13:50:07.978+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:50:07.978+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:07 ERROR SparkContext: Failed to add file:/opt/spark_job/jars/hadoop-aws-3.3.4.jar to Spark environment
[2025-03-26T13:50:07.979+0000] {spark_submit.py:645} INFO - java.io.FileNotFoundException: Jar /opt/spark_job/jars/hadoop-aws-3.3.4.jar not found
[2025-03-26T13:50:07.979+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:2095)
[2025-03-26T13:50:07.980+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2151)
[2025-03-26T13:50:07.980+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.$anonfun$new$15(SparkContext.scala:521)
[2025-03-26T13:50:07.980+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.$anonfun$new$15$adapted(SparkContext.scala:521)
[2025-03-26T13:50:07.981+0000] {spark_submit.py:645} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-26T13:50:07.981+0000] {spark_submit.py:645} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-26T13:50:07.981+0000] {spark_submit.py:645} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-26T13:50:07.982+0000] {spark_submit.py:645} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:521)
[2025-03-26T13:50:07.982+0000] {spark_submit.py:645} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-03-26T13:50:07.983+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-03-26T13:50:07.983+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-03-26T13:50:07.983+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-03-26T13:50:07.984+0000] {spark_submit.py:645} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-03-26T13:50:07.984+0000] {spark_submit.py:645} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-03-26T13:50:07.984+0000] {spark_submit.py:645} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-03-26T13:50:07.985+0000] {spark_submit.py:645} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-26T13:50:07.987+0000] {spark_submit.py:645} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-03-26T13:50:07.988+0000] {spark_submit.py:645} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-03-26T13:50:07.989+0000] {spark_submit.py:645} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-03-26T13:50:07.989+0000] {spark_submit.py:645} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-26T13:50:07.990+0000] {spark_submit.py:645} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-26T13:50:07.990+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:50:08.083+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:08 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-03-26T13:50:08.146+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:08 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2025-03-26T13:50:08.147+0000] {spark_submit.py:645} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2025-03-26T13:50:08.148+0000] {spark_submit.py:645} INFO - at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2025-03-26T13:50:08.148+0000] {spark_submit.py:645} INFO - at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2025-03-26T13:50:08.149+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2025-03-26T13:50:08.149+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2025-03-26T13:50:08.149+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2025-03-26T13:50:08.150+0000] {spark_submit.py:645} INFO - at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2025-03-26T13:50:08.150+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2025-03-26T13:50:08.150+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2025-03-26T13:50:08.151+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2025-03-26T13:50:08.151+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2025-03-26T13:50:08.151+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:50:08.151+0000] {spark_submit.py:645} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/172.18.0.6:7077
[2025-03-26T13:50:08.152+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2025-03-26T13:50:08.152+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2025-03-26T13:50:08.152+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2025-03-26T13:50:08.153+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2025-03-26T13:50:08.153+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2025-03-26T13:50:08.153+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2025-03-26T13:50:08.154+0000] {spark_submit.py:645} INFO - ... 4 more
[2025-03-26T13:50:08.154+0000] {spark_submit.py:645} INFO - Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: spark-master/172.18.0.6:7077
[2025-03-26T13:50:08.154+0000] {spark_submit.py:645} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-03-26T13:50:08.155+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-03-26T13:50:08.155+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-03-26T13:50:08.155+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-03-26T13:50:08.156+0000] {spark_submit.py:645} INFO - at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
[2025-03-26T13:50:08.156+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
[2025-03-26T13:50:08.156+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
[2025-03-26T13:50:08.156+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
[2025-03-26T13:50:08.157+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
[2025-03-26T13:50:08.157+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
[2025-03-26T13:50:08.157+0000] {spark_submit.py:645} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2025-03-26T13:50:08.158+0000] {spark_submit.py:645} INFO - at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2025-03-26T13:50:08.158+0000] {spark_submit.py:645} INFO - at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2025-03-26T13:50:08.158+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:50:28.082+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:28 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-03-26T13:50:28.088+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:28 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2025-03-26T13:50:28.089+0000] {spark_submit.py:645} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2025-03-26T13:50:28.089+0000] {spark_submit.py:645} INFO - at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2025-03-26T13:50:28.089+0000] {spark_submit.py:645} INFO - at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2025-03-26T13:50:28.090+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2025-03-26T13:50:28.090+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2025-03-26T13:50:28.090+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2025-03-26T13:50:28.091+0000] {spark_submit.py:645} INFO - at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2025-03-26T13:50:28.091+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2025-03-26T13:50:28.091+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2025-03-26T13:50:28.092+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2025-03-26T13:50:28.092+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2025-03-26T13:50:28.095+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:50:28.099+0000] {spark_submit.py:645} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/172.18.0.6:7077
[2025-03-26T13:50:28.099+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2025-03-26T13:50:28.100+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2025-03-26T13:50:28.100+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2025-03-26T13:50:28.101+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2025-03-26T13:50:28.101+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2025-03-26T13:50:28.101+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2025-03-26T13:50:28.102+0000] {spark_submit.py:645} INFO - ... 4 more
[2025-03-26T13:50:28.102+0000] {spark_submit.py:645} INFO - Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: spark-master/172.18.0.6:7077
[2025-03-26T13:50:28.103+0000] {spark_submit.py:645} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-03-26T13:50:28.103+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-03-26T13:50:28.103+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-03-26T13:50:28.104+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-03-26T13:50:28.104+0000] {spark_submit.py:645} INFO - at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
[2025-03-26T13:50:28.104+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
[2025-03-26T13:50:28.105+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
[2025-03-26T13:50:28.105+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
[2025-03-26T13:50:28.105+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
[2025-03-26T13:50:28.106+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
[2025-03-26T13:50:28.106+0000] {spark_submit.py:645} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2025-03-26T13:50:28.106+0000] {spark_submit.py:645} INFO - at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2025-03-26T13:50:28.107+0000] {spark_submit.py:645} INFO - at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2025-03-26T13:50:28.107+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:50:48.083+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:48 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-03-26T13:50:48.091+0000] {spark_submit.py:645} INFO - 25/03/26 13:50:48 WARN StandaloneAppClient$ClientEndpoint: Failed to connect to master spark-master:7077
[2025-03-26T13:50:48.094+0000] {spark_submit.py:645} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2025-03-26T13:50:48.099+0000] {spark_submit.py:645} INFO - at org.apache.spark.util.SparkThreadUtils$.awaitResult(SparkThreadUtils.scala:56)
[2025-03-26T13:50:48.101+0000] {spark_submit.py:645} INFO - at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:310)
[2025-03-26T13:50:48.101+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2025-03-26T13:50:48.102+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2025-03-26T13:50:48.103+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2025-03-26T13:50:48.104+0000] {spark_submit.py:645} INFO - at org.apache.spark.deploy.client.StandaloneAppClient$ClientEndpoint$$anon$1.run(StandaloneAppClient.scala:108)
[2025-03-26T13:50:48.106+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[2025-03-26T13:50:48.107+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[2025-03-26T13:50:48.107+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2025-03-26T13:50:48.108+0000] {spark_submit.py:645} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2025-03-26T13:50:48.108+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:50:48.109+0000] {spark_submit.py:645} INFO - Caused by: java.io.IOException: Failed to connect to spark-master/172.18.0.6:7077
[2025-03-26T13:50:48.110+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:294)
[2025-03-26T13:50:48.110+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:214)
[2025-03-26T13:50:48.111+0000] {spark_submit.py:645} INFO - at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:226)
[2025-03-26T13:50:48.111+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.NettyRpcEnv.createClient(NettyRpcEnv.scala:204)
[2025-03-26T13:50:48.111+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:202)
[2025-03-26T13:50:48.112+0000] {spark_submit.py:645} INFO - at org.apache.spark.rpc.netty.Outbox$$anon$1.call(Outbox.scala:198)
[2025-03-26T13:50:48.113+0000] {spark_submit.py:645} INFO - ... 4 more
[2025-03-26T13:50:48.113+0000] {spark_submit.py:645} INFO - Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: spark-master/172.18.0.6:7077
[2025-03-26T13:50:48.114+0000] {spark_submit.py:645} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-03-26T13:50:48.114+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-03-26T13:50:48.114+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-03-26T13:50:48.115+0000] {spark_submit.py:645} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-03-26T13:50:48.115+0000] {spark_submit.py:645} INFO - at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337)
[2025-03-26T13:50:48.116+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
[2025-03-26T13:50:48.117+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
[2025-03-26T13:50:48.118+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
[2025-03-26T13:50:48.118+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
[2025-03-26T13:50:48.119+0000] {spark_submit.py:645} INFO - at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
[2025-03-26T13:50:48.119+0000] {spark_submit.py:645} INFO - at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[2025-03-26T13:50:48.119+0000] {spark_submit.py:645} INFO - at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[2025-03-26T13:50:48.120+0000] {spark_submit.py:645} INFO - at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
[2025-03-26T13:50:48.120+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:51:08.082+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 WARN StandaloneSchedulerBackend: Application ID is not initialized yet.
[2025-03-26T13:51:08.083+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 ERROR StandaloneSchedulerBackend: Application has been killed. Reason: All masters are unresponsive! Giving up.
[2025-03-26T13:51:08.091+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 43805.
[2025-03-26T13:51:08.092+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO NettyBlockTransferService: Server created on 9d8a997eda77:43805
[2025-03-26T13:51:08.092+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-03-26T13:51:08.098+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 9d8a997eda77, 43805, None)
[2025-03-26T13:51:08.103+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO BlockManagerMasterEndpoint: Registering block manager 9d8a997eda77:43805 with 434.4 MiB RAM, BlockManagerId(driver, 9d8a997eda77, 43805, None)
[2025-03-26T13:51:08.105+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 9d8a997eda77, 43805, None)
[2025-03-26T13:51:08.106+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 9d8a997eda77, 43805, None)
[2025-03-26T13:51:08.282+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-03-26T13:51:08.717+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-03-26T13:51:08.720+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO SharedState: Warehouse path is 'file:/opt/spark-warehouse'.
[2025-03-26T13:51:08.783+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-03-26T13:51:08.794+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO SparkUI: Stopped Spark web UI at http://9d8a997eda77:4040
[2025-03-26T13:51:08.808+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO StandaloneSchedulerBackend: Shutting down all executors
[2025-03-26T13:51:08.811+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2025-03-26T13:51:08.820+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 WARN StandaloneAppClient$ClientEndpoint: Drop UnregisterApplication(null) because has not yet connected to master
[2025-03-26T13:51:08.829+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-03-26T13:51:08.843+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO MemoryStore: MemoryStore cleared
[2025-03-26T13:51:08.845+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO BlockManager: BlockManager stopped
[2025-03-26T13:51:08.854+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-03-26T13:51:08.857+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-03-26T13:51:08.866+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:08 INFO SparkContext: Successfully stopped SparkContext
[2025-03-26T13:51:10.224+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:10 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: s3a://raw/playlist_20250326_133348.json.
[2025-03-26T13:51:10.225+0000] {spark_submit.py:645} INFO - java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2025-03-26T13:51:10.226+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2688)
[2025-03-26T13:51:10.226+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3431)
[2025-03-26T13:51:10.227+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3466)
[2025-03-26T13:51:10.227+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
[2025-03-26T13:51:10.228+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
[2025-03-26T13:51:10.228+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
[2025-03-26T13:51:10.228+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
[2025-03-26T13:51:10.229+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
[2025-03-26T13:51:10.229+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:53)
[2025-03-26T13:51:10.230+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:366)
[2025-03-26T13:51:10.230+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2025-03-26T13:51:10.230+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2025-03-26T13:51:10.231+0000] {spark_submit.py:645} INFO - at scala.Option.getOrElse(Option.scala:189)
[2025-03-26T13:51:10.231+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2025-03-26T13:51:10.231+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:362)
[2025-03-26T13:51:10.232+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-03-26T13:51:10.232+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-03-26T13:51:10.233+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-03-26T13:51:10.233+0000] {spark_submit.py:645} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-03-26T13:51:10.234+0000] {spark_submit.py:645} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-03-26T13:51:10.234+0000] {spark_submit.py:645} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-26T13:51:10.235+0000] {spark_submit.py:645} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2025-03-26T13:51:10.235+0000] {spark_submit.py:645} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-03-26T13:51:10.235+0000] {spark_submit.py:645} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-03-26T13:51:10.236+0000] {spark_submit.py:645} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-26T13:51:10.236+0000] {spark_submit.py:645} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-26T13:51:10.237+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:51:10.238+0000] {spark_submit.py:645} INFO - Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2025-03-26T13:51:10.238+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2592)
[2025-03-26T13:51:10.239+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2686)
[2025-03-26T13:51:10.239+0000] {spark_submit.py:645} INFO - ... 26 more
[2025-03-26T13:51:10.291+0000] {spark_submit.py:645} INFO - Traceback (most recent call last):
[2025-03-26T13:51:10.291+0000] {spark_submit.py:645} INFO - File "/spark_job/data_transformation.py", line 43, in <module>
[2025-03-26T13:51:10.293+0000] {spark_submit.py:645} INFO - data_transformation(spark, file_path)
[2025-03-26T13:51:10.294+0000] {spark_submit.py:645} INFO - File "/spark_job/data_transformation.py", line 20, in data_transformation
[2025-03-26T13:51:10.296+0000] {spark_submit.py:645} INFO - df_raw = spark.read.option("multiline", "true").json(file_path)#"s3a://raw/playlist_20250323_190824.json")
[2025-03-26T13:51:10.297+0000] {spark_submit.py:645} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-03-26T13:51:10.297+0000] {spark_submit.py:645} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 425, in json
[2025-03-26T13:51:10.298+0000] {spark_submit.py:645} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-03-26T13:51:10.298+0000] {spark_submit.py:645} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2025-03-26T13:51:10.299+0000] {spark_submit.py:645} INFO - File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-03-26T13:51:10.299+0000] {spark_submit.py:645} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o35.json.
[2025-03-26T13:51:10.300+0000] {spark_submit.py:645} INFO - : java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2025-03-26T13:51:10.300+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2688)
[2025-03-26T13:51:10.301+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3431)
[2025-03-26T13:51:10.301+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3466)
[2025-03-26T13:51:10.301+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem.access$300(FileSystem.java:174)
[2025-03-26T13:51:10.302+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3574)
[2025-03-26T13:51:10.302+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3521)
[2025-03-26T13:51:10.303+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:540)
[2025-03-26T13:51:10.303+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
[2025-03-26T13:51:10.304+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$1(DataSource.scala:724)
[2025-03-26T13:51:10.304+0000] {spark_submit.py:645} INFO - at scala.collection.immutable.List.map(List.scala:293)
[2025-03-26T13:51:10.305+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.checkAndGlobPathIfNecessary(DataSource.scala:722)
[2025-03-26T13:51:10.305+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.execution.datasources.DataSource.checkAndGlobPathIfNecessary(DataSource.scala:551)
[2025-03-26T13:51:10.306+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:404)
[2025-03-26T13:51:10.306+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2025-03-26T13:51:10.307+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2025-03-26T13:51:10.307+0000] {spark_submit.py:645} INFO - at scala.Option.getOrElse(Option.scala:189)
[2025-03-26T13:51:10.307+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2025-03-26T13:51:10.308+0000] {spark_submit.py:645} INFO - at org.apache.spark.sql.DataFrameReader.json(DataFrameReader.scala:362)
[2025-03-26T13:51:10.308+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-03-26T13:51:10.309+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-03-26T13:51:10.309+0000] {spark_submit.py:645} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-03-26T13:51:10.310+0000] {spark_submit.py:645} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-03-26T13:51:10.310+0000] {spark_submit.py:645} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-03-26T13:51:10.311+0000] {spark_submit.py:645} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-26T13:51:10.311+0000] {spark_submit.py:645} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2025-03-26T13:51:10.311+0000] {spark_submit.py:645} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-03-26T13:51:10.312+0000] {spark_submit.py:645} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-03-26T13:51:10.312+0000] {spark_submit.py:645} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-26T13:51:10.313+0000] {spark_submit.py:645} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-26T13:51:10.313+0000] {spark_submit.py:645} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-03-26T13:51:10.313+0000] {spark_submit.py:645} INFO - Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
[2025-03-26T13:51:10.314+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2592)
[2025-03-26T13:51:10.314+0000] {spark_submit.py:645} INFO - at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2686)
[2025-03-26T13:51:10.315+0000] {spark_submit.py:645} INFO - ... 29 more
[2025-03-26T13:51:10.315+0000] {spark_submit.py:645} INFO - 
[2025-03-26T13:51:10.360+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:10 INFO ShutdownHookManager: Shutdown hook called
[2025-03-26T13:51:10.360+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:10 INFO ShutdownHookManager: Deleting directory /tmp/spark-45b236ee-3255-432b-947d-888b71f04559
[2025-03-26T13:51:10.366+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:10 INFO ShutdownHookManager: Deleting directory /tmp/spark-b92550c5-7fb0-4db8-8000-c9109ac36dd0
[2025-03-26T13:51:10.371+0000] {spark_submit.py:645} INFO - 25/03/26 13:51:10 INFO ShutdownHookManager: Deleting directory /tmp/spark-b92550c5-7fb0-4db8-8000-c9109ac36dd0/pyspark-d7b4676e-f53e-487d-b0ee-3c4ec0d8f53f
[2025-03-26T13:51:10.407+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-03-26T13:51:10.419+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/etl_pipeline.py", line 96, in spark_dtransformation
    ).execute(kwargs)
      ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 567, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=2 --conf spark.jars=/opt/spark_job/aws-java-sdk-bundle-1.12.262.jar,/opt/spark_job/jars/hadoop-aws-3.3.4.jar --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=admin --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem --conf spark.hadoop.fs.s3a.path.style.access=true --principal *** --name arrow-spark --verbose --deploy-mode client /spark_job/data_transformation.py s3://raw/playlist_20250326_133348.json. Error code is: 1.
[2025-03-26T13:51:10.427+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=etl_pipeline, task_id=transformation, run_id=manual__2025-03-26T13:33:39.761724+00:00, execution_date=20250326T133339, start_date=20250326T135002, end_date=20250326T135110
[2025-03-26T13:51:10.471+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 9 for task transformation (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=2 --conf spark.jars=/opt/spark_job/aws-java-sdk-bundle-1.12.262.jar,/opt/spark_job/jars/hadoop-aws-3.3.4.jar --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=admin --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem --conf spark.hadoop.fs.s3a.path.style.access=true --principal *** --name arrow-spark --verbose --deploy-mode client /spark_job/data_transformation.py s3://raw/playlist_20250326_133348.json. Error code is: 1.; 1120)
[2025-03-26T13:51:10.485+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-03-26T13:51:10.509+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-26T13:51:10.545+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
